### ISSUE TITLE
DTBook content leads to I/O error (and failed validation outcome)

### ISSUE THREAD

**Issue by bitsgalore** [IDPF/epubcheck#518]
While trying to validate [this EPUB 2 file with DTBook content](https://github.com/KBNLresearch/epubPolicyTests/blob/master/build/epub20_dtbook.epub?raw=true) Epubcheck 3.0.1 gave me this error:
 
    ERROR: : I/O error reading OEBPS/hauy-2005-1.xml: Stream closed
    
And similarly for *Epubcheck* 4:
 
    RSC-005, ERROR, [Error while parsing file 'value of attribute "version" is invalid; must be equal to "2005-2"'.], OEBPS/hauy-2005-1.xml (4-88)

This then causes the validation to fail. The [source of that EPUB is here](https://github.com/KBNLresearch/epubPolicyTests/tree/master/content/epub20_dtbook). The offending  *hauy-2005-1.xml* file resource is valid against the [*DTBook* DTD](http://www.daisy.org/z3986/2005/dtbook-2005-1.dtd) (I checked this using both [JHOVE](http://jhove.sourceforge.net/) and an [online XML validator](http://www.validome.org/xml/validate/)). Another *EPUB* with a *DTBook* resource gave me the same error. So it looks like *Epubcheck* doesn't properly handle *DTBook* content.

**Comment by mattgarrish** [comment]
The epubcheck error is correct.

EPUB 2 allows for content that conforms to the 2005-2 version of DTBook, not 2005-1. See the manifest section: http://www.idpf.org/epub/20/spec/OPF_2.0.1_draft.htm#Section2.3

and if you open the DTD for 2005-2, it’s explicit that that value be used for the version:

<!ATTLIST dtbook
  version CDATA #FIXED '2005-2'

Your source is in error, in other words.

Matt

From: Johan van der Knijff 
Sent: Friday, March 13, 2015 12:35 PM
To: IDPF/epubcheck 
Subject: [epubcheck] DTBook content leads to I/O error (and failed validation outcome) (#518)

While trying to validate this EPUB 2 file with DTBook content Epubcheck 3.0.1 gave me this error:

ERROR: : I/O error reading OEBPS/hauy-2005-1.xml: Stream closed
And similarly for Epubcheck 4:

RSC-005, ERROR, [Error while parsing file 'value of attribute "version" is invalid; must be equal to "2005-2"'.], OEBPS/hauy-2005-1.xml (4-88)
This then causes the validation to fail. The source of that EPUB is here. The offending hauy-2005-1.xml file resource is valid against the DTBook DTD (I checked this using both JHOVE and an online XML validator). Another EPUB with a DTBook resource gave me the same error. So it looks like Epubcheck doesn't properly handle DTBook content.

**Comment by bitsgalore** [comment]
Hi Matt,

You're right, I hadn't thought of that. That made me wonder about another one I had been using in some earlier tests:

<http://robertjszmidt.pl/sites/default/files/CG_RJS-dtbook.epub>

This epub *is* using the correct version of the DTD:

    <!DOCTYPE dtbook PUBLIC "-//NISO//DTD dtbook 2005-2//EN" "http://www.daisy.org/z3986/2005/dtbook-2005-2.dtd">

The XML also validates against the DTD; nevertheless if I throw it at epubcheck 3.0.1 it still gives the same I/O error. The 4.0.0. alpha version doesn't give the error and behaves correctly, so this looks like this is fixed in the latest release.

**Comment by mattgarrish** [comment]
Right, I guess that shows how often people are validating dtbook in epubcheck. The format never really took off, as you know. I can’t speak to what might have broken, but at least it’s working again.

And for the record, I pointed to the manifest dtd list after a quick search, but the formal requirement for 2005-2 compliance is in the ops spec: http://www.idpf.org/epub/20/spec/OPS_2.0.1_draft.htm#Section1.4.1.3

Matt

**Comment by bitsgalore** [comment]
Yes, it did take me some effort to find DTBook examples and they appear to be pretty rare. The main reason I'm interested in them is this:

<https://researchkb.wordpress.com/2015/03/13/policy-based-assessment-of-epub-with-epubcheck/> 

BTW to avoid further confusion I updated the DTD reference in my original example file to 2005-2, just in case other people end up re-using it. I also corrected the blog accordingly. I'll close the issue now, as v 4 is obviously showing the correct behavior.

Thanks again!

Johan
